src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -37,17 +37,23 @@
 #include "unit-name.h"
 #include "virt.h"
 
+typedef enum GptAutoRoot {
+        GPT_AUTO_ROOT_UNSPECIFIED,  /* no root= specified */
+        GPT_AUTO_ROOT_OFF,          /* root= set to something else */
+        GPT_AUTO_ROOT_ON,           /* root= set explicitly to "gpt-auto" */
+        GPT_AUTO_ROOT_FORCE,        /* root= set explicitly to "gpt-auto-force" → ignores factory reset mode */
+} GptAutoRoot;
+
```

Comment:
I'd prefer to move this to shared/generator.c with a generic parser (parse_gpt_auto_root(), where _OFF is mapped to 0), and make fstab_generator check parse_gpt_auto_root() > 0 to determine whether gpt auto is specified.